server/services/archive_service.py

```
@@ -0,0 +1,145 @@
+        """Archive a disaster and its related data."""
+        async with self.db.transaction():
+            try:
+                # Archive the disaster
+                await self._archive_disaster_record(disaster_id)
+                # Archive related posts
+                await self._archive_disaster_posts(disaster_id)
+                # Archive related alerts
+                await self._archive_disaster_alerts(disaster_id)
+                # Mark original disaster as archived
+                await self._mark_disaster_archived(disaster_id)
+                return True
```

Ruff (0.14.0)

Consider moving this statement to an `else` block

(TRY300)